client/packages/core/src/queryTypes.ts

```diff
@@ -10,6 +10,12 @@ import type {
   ResolveEntityAttrs,
 } from "./schemaTypes";
 
+type Expand<T> = T extends object
```

Comment:
I originally had a more fully-featured version of this, which handled non-objects too. See: 

https://github.com/sindresorhus/type-fest/blob/main/source/simplify-deep.d.ts

However, I think it's actually better to _just_ handle objects.[ I chose to use the 4 liner from jcatz](https://stackoverflow.com/a/57683652/1292663)

My main reason:

This maps our types. It _should_ be a no-op for TS, but the more complicated we make it, the more likely that it in fact is not a no-op.